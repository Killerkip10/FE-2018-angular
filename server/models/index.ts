export {User} from './user';
export {Login} from './login';
export {ShortUser} from './short-user';
