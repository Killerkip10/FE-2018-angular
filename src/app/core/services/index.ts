export * from './auth.service';
export * from './profile.service';
export * from './token.service';
export * from './search-user.service';
