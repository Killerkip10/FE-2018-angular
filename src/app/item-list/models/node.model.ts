export interface Node {
  value: string;
  open: boolean;
  children: Node[];
}
