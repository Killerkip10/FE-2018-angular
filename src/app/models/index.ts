export * from '../../../server/models/index';
