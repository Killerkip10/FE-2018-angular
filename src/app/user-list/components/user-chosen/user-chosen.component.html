<div class="chosen">
  <div class="container">
    <div>{{user?.name}}</div>
    <div>{{user?.login}}</div>
    <div>{{user?.dateOfBirth | date: 'medium'}}</div>
  </div>

  <mat-icon>keyboard_arrow_down</mat-icon>
</div>
