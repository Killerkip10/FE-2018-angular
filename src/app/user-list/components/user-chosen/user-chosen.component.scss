.chosen{
  min-height: 70px;
  display: flex;
  align-items: center;
  .container{
    padding: 5px;
    flex: 1;
  }
  mat-icon{
    width: 30px;
    height: 30px;
    font-size: 30px;
  }
}
