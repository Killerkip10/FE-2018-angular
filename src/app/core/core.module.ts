import {NgModule} from '@angular/core';
import {BrowserModule} from '@angular/platform-browser';
import {TranslateModule} from '@ngx-translate/core';
import {HttpClientModule} from '@angular/common/http';
import {RouterModule} from '@angular/router';

import {BrowserAnimationsModule} from '@angular/platform-browser/animations';
import {AuthGuard} from './guards';
import {AuthService} from './services';
import {
  MatButtonModule,
  MatToolbarModule,
  MatSelectModule,
  MatIconModule
} from '@angular/material';
import {
  HeaderComponent,
  FooterComponent
} from './components';

@NgModule({
  imports: [
    BrowserModule,
    HttpClientModule,
    TranslateModule,
    BrowserAnimationsModule,

    RouterModule,

    MatIconModule,
    MatButtonModule,
    MatSelectModule,
    MatToolbarModule
  ],
  declarations: [
    HeaderComponent,
    FooterComponent
  ],
  providers: [
    AuthService,

    AuthGuard
  ],
  exports: [
    HeaderComponent,
    FooterComponent
  ]
})
export class CoreModule {}
