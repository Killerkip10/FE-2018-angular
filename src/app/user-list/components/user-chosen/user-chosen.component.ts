import {Component, Input} from '@angular/core';

import {User} from '../../../models';

@Component({
  selector: 'app-user-chosen',
  templateUrl: './user-chosen.component.html',
  styleUrls: ['./user-chosen.component.scss']
})
export class UserChosenComponent {
  @Input() user: User;

  constructor() { }
}
