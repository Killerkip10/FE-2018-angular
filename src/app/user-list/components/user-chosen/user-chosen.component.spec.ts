import {async, ComponentFixture, TestBed} from '@angular/core/testing';
import {NO_ERRORS_SCHEMA} from '@angular/core';

import {UserChosenComponent} from './user-chosen.component';
import {User} from '../../../models';

describe('UserChosenComponent', () => {
  let component: UserChosenComponent;
  let fixture: ComponentFixture<UserChosenComponent>;

  beforeEach(async(() => {
    TestBed.configureTestingModule({
      declarations: [
        UserChosenComponent
      ],
      schemas: [
        NO_ERRORS_SCHEMA
      ]
    }).compileComponents();
  }));

  beforeEach(() => {
    fixture = TestBed.createComponent(UserChosenComponent);
    component = fixture.componentInstance;
    fixture.detectChanges();
  });

  it('should create', () => {
    expect(component).toBeTruthy();
  });

  it('snapshot', () => {
    expect(fixture).toMatchSnapshot();
  });

  it('snapshot', () => {
    component.user = {} as User;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });

  it('snapshot', () => {
    component.user = {
      login: '1234',
      name: 'Pasha',
      dateOfBirth: '2018-11-10T18:52:23.023Z'
    } as User;
    fixture.detectChanges();

    expect(fixture).toMatchSnapshot();
  });
});
