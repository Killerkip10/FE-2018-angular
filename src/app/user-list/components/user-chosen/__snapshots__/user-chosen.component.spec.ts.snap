// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`UserChosenComponent snapshot 1`] = `
<app-user-chosen>
  <div
    class="chosen"
  >
    <div
      class="container"
    >
      <div>
        
      </div>
      <div>
        
      </div>
      <div>
        
      </div>
    </div>
    <mat-icon>
      keyboard_arrow_down
    </mat-icon>
  </div>
</app-user-chosen>
`;

exports[`UserChosenComponent snapshot 2`] = `
<app-user-chosen
  user={[Function Object]}
>
  <div
    class="chosen"
  >
    <div
      class="container"
    >
      <div>
        
      </div>
      <div>
        
      </div>
      <div>
        
      </div>
    </div>
    <mat-icon>
      keyboard_arrow_down
    </mat-icon>
  </div>
</app-user-chosen>
`;

exports[`UserChosenComponent snapshot 3`] = `
<app-user-chosen
  user={[Function Object]}
>
  <div
    class="chosen"
  >
    <div
      class="container"
    >
      <div>
        Pasha
      </div>
      <div>
        1234
      </div>
      <div>
        Nov 10, 2018, 9:52:23 PM
      </div>
    </div>
    <mat-icon>
      keyboard_arrow_down
    </mat-icon>
  </div>
</app-user-chosen>
`;
